tests/test_testRepo.py

```diff
@@ -80,21 +81,38 @@ def _checkButlerDimension(self, dimensions, query, expected):
         self.assertIn(dict(result[0]), expected)
 
     def testButlerDimensions(self):
-        self. _checkButlerDimension({"instrument"},
-                                    "instrument='notACam'",
-                                    [{"instrument": "notACam"}, {"instrument": "dummyCam"}])
-        self. _checkButlerDimension({"visit", "instrument"},
-                                    "visit=101",
-                                    [{"instrument": "notACam", "visit": 101},
-                                     {"instrument": "dummyCam", "visit": 101}])
-        self. _checkButlerDimension({"visit", "instrument"},
-                                    "visit=102",
-                                    [{"instrument": "notACam", "visit": 102},
-                                     {"instrument": "dummyCam", "visit": 102}])
-        self. _checkButlerDimension({"detector", "instrument"},
-                                    "detector=5",
-                                    [{"instrument": "notACam", "detector": 5},
-                                     {"instrument": "dummyCam", "detector": 5}])
+        self._checkButlerDimension({"instrument"},
+                                   "instrument='notACam'",
+                                   [{"instrument": "notACam"}, {"instrument": "dummyCam"}])
+        self._checkButlerDimension({"visit", "instrument"},
+                                   "visit=101",
+                                   [{"instrument": "notACam", "visit": 101},
+                                    {"instrument": "dummyCam", "visit": 101}])
+        self._checkButlerDimension({"visit", "instrument"},
+                                   "visit=102",
+                                   [{"instrument": "notACam", "visit": 102},
+                                    {"instrument": "dummyCam", "visit": 102}])
+        self._checkButlerDimension({"detector", "instrument"},
+                                   "detector=5",
+                                   [{"instrument": "notACam", "detector": 5},
+                                    {"instrument": "dummyCam", "detector": 5}])
+
+    def testAddDataIdValue(self):
```

Comment:
Possibly missing: cases where a dimension and value are given, but necessary linked additional dimensions and values are **not** given.